### PR TITLE
update to file naming and structure: bambu-processReads and utility functions

### PR DESCRIPTION
**Pull request by jonathangoeke**
- created utility function files for bambu.processReads according to the major tasks (junction table creation, error correction, read class construction).
- seqlevel consistency checks are now performed based on the reference genome sequence that is provided. seqlevels and ranges which are not present in the genome sequence are dropped.
- bugfix: strand correction for reads now corrects for reads (previously was likely to be disabled)

### CODE REVIEW

File: R/bambu-processReads.R [hunk 1]
```diff
@@ -53,20 +53,49 @@ bambu.processReads <- function(reads, readClass.file, annotations, genomeSequenc
 #' @noRd
 bambu.processReadsByFile <- function(bam.file, genomeSequence, annotations,
```
Comment (cying111) [inline R/bambu-processReads.R:54]: This function seems to have a length of over 50 lines? Maybe can slightly reduce it a bit?

File: R/bambu-processReads_utilityCreateJunctionTables.R [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+    mcols(gr)$id <- rep(mcols(x)$id, elementNROWS(gaps))
+  return(gr)
+}
+
+
+
+#' Create Junction tables from unlisted junction granges
+#' @importFrom BiocParallel bppram bpvec
+#' @noRd
+createJunctionTable <- function(unlisted_junctions,
```
Comment (cying111) [inline R/bambu-processReads_utilityCreateJunctionTables.R:71]: This function also has a length slightly over 50 lines?

**Review by cying111 — CHANGES_REQUESTED** [review]
Hi @jonathangoeke, thanks for the updates, the changes look fine to me, only two functions have slightly length over 50 lines, maybe can consider to shorten the two functions a bit?

File: R/bambu-processReads.R (hunk 1 above)
Comment (jonathangoeke) [reply R/bambu-processReads.R:54]: the function is now less than 50 lines

File: R/bambu-processReads_utilityCreateJunctionTables.R (hunk 2 above)
Comment (jonathangoeke) [reply R/bambu-processReads_utilityCreateJunctionTables.R:71]: I removed a check for constent seqlevels which was redundant, the function is now below 50 lines

**Comment by jonathangoeke** [comment]
Thanks @cying111 for looking through this. I changed the 2 functions to reduce the line number

**Review by cying111 — APPROVED** [review]

**Review by Drakyrie — APPROVED** [review]

**Pull request merged**